src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1233,7 +1234,7 @@ private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
 
 #if DEBUG
             // https://github.com/dotnet/roslyn/issues/34993 Enable for all calls
-            if (compilation.NullableSemanticAnalysisEnabled)
+            if (compilation.IsNullableAnalysisEnabled)
```

Comment:
Why is this not `!compilation.IsNullableAnalysisExplicitlyDisabled`? The callers of this method use that form of the check. Seems like we're inconsistently making decisions on this code path. Am I missing something about this part of the code that means we shouldn't run it here?